invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/Profile/components/PatronCurrentLoans/PatronCurrentLoans.js

```diff
@@ -2,37 +2,42 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import isEmpty from 'lodash/isEmpty';
 import pick from 'lodash/pick';
-import { Button } from 'semantic-ui-react';
 
-import { invenioConfig, ES_MAX_SIZE } from '../../../../../common/config';
-import { loan as loanApi } from '../../../../../common/api/';
-import { Loader, Error, ResultsTable } from '../../../../../common/components';
+import {
+  Loader,
+  Error,
+  ResultsTable,
+  Pagination,
+} from '../../../../../common/components';
 import { formatter } from '../../../../../common/components/ResultsTable/formatters';
+import { DEFAULT_RESULTS_SIZE } from '../../../../../common/config';
 
 export default class PatronCurrentLoans extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchPatronCurrentLoans = this.props.fetchPatronCurrentLoans;
+    this.patronPid = this.props.patronPid;
+    this.state = { activePage: 1 };
+  }
+
   componentDidMount() {
-    const { patronPid } = this.props;
-    this.props.fetchPatronCurrentLoans(patronPid);
+    this.fetchPatronCurrentLoans(this.patronPid);
   }
 
-  handleShowAllClick() {
-    const { patronPid } = this.props;
-    const query = loanApi
-      .query()
-      .withPatronPid(patronPid)
-      .withState(invenioConfig.circulation.loanActiveStates)
-      .withSize(ES_MAX_SIZE)
-      .sortByNewest()
-      .qs();
-    this.showMaxRows = ES_MAX_SIZE;
-    this.props.fetchPatronCurrentLoans(patronPid, query);
+  updatePage(activePage) {
+    this.fetchPatronCurrentLoans(this.patronPid, activePage);
+    this.setState({ activePage: activePage });
   }
 
-  showAllButton = () => {
+  paginationComponent = () => {
     return (
-      <Button size="tiny" onClick={() => this.handleShowAllClick()}>
-        Show all
-      </Button>
+      <Pagination
+        currentPage={this.state.activePage}
+        currentSize={DEFAULT_RESULTS_SIZE}
+        loading={this.props.isLoading}
+        totalResults={this.props.data.total}
+        updatePage={this.updatePage.bind(this)}
```

Comment:
I think we don't need this `bind` if you change the `updatePage` function to an arrow function